rsconnect/validation.py

```diff
@@ -1,46 +1,119 @@
 import typing
 
+import click
+
 from rsconnect.exception import RSConnectException
 
 
-def _get_present_options(options: typing.Dict[str, typing.Optional[str]]) -> typing.List[str]:
-    return [k for k, v in options.items() if v]
+def _get_present_options(
+    options: typing.Dict[str, typing.Optional[typing.Any]],
+    ctx: click.Context = None,
+) -> typing.List[str]:
+    result: typing.List[str] = []
+    for k, v in options.items():
+        if v:
+            parts = k.split("--")
+            if ctx and len(parts) == 2:
+                source = ctx.get_parameter_source(parts[1]).name  # type: ignore
```

Comment:
It looks like ctx.get_parameter_source(parts[1]) returns None in some cases according to the @kgartland-rstudio 's bug report.